File: chroma-league/src/test/java/net/booone/chromaleague/rest/eventdata/EventTest.java
package net.booone.chromaleague.rest.eventdata;

import com.jsoniter.JsonIterator;
import net.booone.chromaleague.rest.http.LeagueHttpClientMock;
import org.junit.jupiter.api.Test;

import java.io.File;
import java.io.IOException;
import java.net.URISyntaxException;
import java.nio.file.Files;
import java.util.List;
import java.util.Objects;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

class EventTest {
    @Test
    void testArtificialEventRead() {
        //given
        final String json = readJsonResource("fullevent.json");

        //when
        final Event baseEvent = JsonIterator.deserialize(json, Event.class);

        //then
        assertEquals(7, baseEvent.EventID());
        assertEquals("DragonKill", baseEvent.EventName());
        assertEquals(933.6826782226563, baseEvent.EventTime());
        assertEquals("Fire", baseEvent.DragonType());
        assertEquals("BooonE", baseEvent.KillerName());
        assertEquals("Noob", baseEvent.VictimName());
        assertEquals("Win", baseEvent.Result());
        assertTrue(baseEvent.Assisters().containsAll(List.of("Teammate 1", "Teammate 2")));
    }

    private static String readJsonResource(String resourceName) {
        try {
            return Files.readString(
                    new File(Objects.requireNonNull(LeagueHttpClientMock.class.getClassLoader()
                            .getResource("json/" + resourceName)).toURI()).toPath());
        } catch (IOException | URISyntaxException e) {
            throw new AssertionError(e);
        }
    }
}

File: test/rest/eventdata/Event.test.ts
import {describe, expect, test} from 'vitest';
import type {Event} from '../../../src/rest/eventdata/Event.js';

describe('EventTest', () => {
    test('testArtificialEventRead', () => {
        //given
        const json: Event = {
            Assisters: ['Teammate 1', 'Teammate 2'],
            DragonType: 'Fire',
            EventID: 7,
            EventName: 'DragonKill',
            EventTime: 933.6826782226563,
            KillerName: 'BooonE',
            VictimName: 'Noob',
            Result: 'Win',
        };

        //then
        expect(json.EventID).toBe(7);
        expect(json.EventName).toBe('DragonKill');
        expect(json.EventTime).toBe(933.6826782226563);
        expect(json.DragonType).toBe('Fire');
        expect(json.KillerName).toBe('BooonE');
        expect(json.VictimName).toBe('Noob');
        expect(json.Result).toBe('Win');
        expect(json.Assisters).toContain('Teammate 1');
        expect(json.Assisters).toContain('Teammate 2');
    });
});
